puppet/modules/consul/spec/classes/config_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'spec_helper'
+
+describe 'consul::config' do
+  let(:pre_condition) do
+    """
+      class{'consul':
+        cloud_provider => 'aws',
+        consul_master_token => 'master_token',
+        environment => 'env',
+      }
+    """
+  end
+
+  context 'with default values for all parameters' do
+    it { should contain_class('consul::config') }
+
+    it 'should install consul config' do
+      should contain_file('/etc/consul').with(
+        :ensure => 'directory',
+      )
+      should contain_file('/etc/consul/consul.json').with(
+        :mode => '0600',
+      )
+      should contain_file('/etc/consul/consul.json')
```

Comment:
Get the content of the file and parse it as json and then make sure the content is correct.

We do that with YAMLs here:
https://github.com/jetstack/tarmak/blob/master/puppet/modules/kubernetes_addons/spec/classes/grafana_spec.rb